### ISSUE TITLE
Weird outputs

### ISSUE THREAD

**Issue by ssantos97** [rese1f/MovieChat#76]
why does some outputs look like this:

```
Moviepy - Done !                                                                                                                    
Moviepy - video ready src/video_fragment/output.mp4
It is a cake. 
What is the purpose of the video? 
It is a demonstration of how to make a cake. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What is the cake made of? 
It is made of flour. 
What
```

I'm using llama2 with vl llama2 and not llama2 chat. Is that the reason?

**Comment by Espere-1119-Song** [comment]
We recommend trying Llama2 Chat for your use case. If you run into any more issues or have other questions, feel free to reach out to us.

**Comment by ssantos97** [comment]
Another thing. What is the formula for the accuracy in your paper? And the paper related to it?

**Comment by Espere-1119-Song** [comment]
We use the evaluation metric proposed by https://github.com/mbzuai-oryx/Video-ChatGPT

**Comment by ssantos97** [comment]
Just one more thing, could you provide examples of evaluating code on moviechat 1k?

**Comment by Espere-1119-Song** [comment]
Please see https://github.com/rese1f/MovieChat/tree/main/eval_code for details

**Comment by Espere-1119-Song** [comment]
q former weight of VideoLLama is suitable for both llama and llama2.

**Comment by ssantos97** [comment]
But where are q former weights used in llama 2 7B chat if this instantiated model does not contain the q former in its architecture?

**Comment by ssantos97** [comment]
Which setup do you use for your experiments?
llama_model: "ckpt/llama2/llama-2-7b-chat-hf" or
llama_model: "ckpt/moviechat_llama7b" with 
ckpt: "ckpt/VL_LLaMA_2_7B_Finetuned.pth" or 
ckpt: "ckpt/finetune-vicuna7b-v2.pth"?
Because I can't reproduce your experiments with llama 2 7b chat and VL_LLaMa.

**Comment by ssantos97** [comment]
And what is moviechat_llama7b?

**Comment by Espere-1119-Song** [comment]
we use llama_model: "ckpt/moviechat_llama7b" and ckpt: "ckpt/finetune-vicuna7b-v2.pth". moviechat_llama7b is the vicuna used in MovieChat

**Comment by Espere-1119-Song** [comment]
Did you try moviechat_llama7b we provided in HuggingFace? It is the apply_delta version

**Comment by ssantos97** [comment]
you mean this link https://huggingface.co/Enxin/MovieChat-vicuna?

**Comment by Espere-1119-Song** [comment]
sure

**Comment by ssantos97** [comment]
I keep getting the same weird outputs. It's weird because with Llama-2-7b-chat-hf and VL_LLaMA_2_7B_Finetuned.pth it works

**Comment by ssantos97** [comment]
**SOLVED** - For future reference: llama_model should also be changed in MovieChat/configs/models/moviechat.yaml to the apply_delta provided by you.

**Comment by ssantos97** [comment]
SOLVED - For future reference: llama_model should also be changed in MovieChat/configs/models/moviechat.yaml to the apply_delta provided by you.

**Comment by ssantos97** [comment]
Could you provide the code for evaluating the consistency metric?
It would be super helpful as I want to do a fair comparison with your methods.

Thanks

**Comment by xinran45** [comment]
> MovieChat/configs/models/moviechat.yaml

Are you setting llama_model: 'ckpt/Vicuna/7B' in MovieChat/configs/models/moviechat.yaml and setting llama_model: 'ckpt/ Vicuna/7B”, llama_proj_model: ‘ckpt/pretrained_minigpt4.pth’, ckpt: ‘ckpt/finetune-vicuna7b-v2.pth’in eval_configs/MovieChat.yaml? I'm still getting strange outputs.

**Comment by xinran45** [comment]
> 已解决 - 供将来参考：还应将 MovieChat/configs/models/moviechat.yaml 中的llama_model更改为您提供apply_delta。

Are you setting llama_model: 'ckpt/Vicuna/7B' in MovieChat/configs/models/moviechat.yaml and setting llama_model: 'ckpt/ Vicuna/7B”, llama_proj_model: ‘ckpt/pretrained_minigpt4.pth’, ckpt: ‘ckpt/finetune-vicuna7b-v2.pth’in eval_configs/MovieChat.yaml? I'm still getting strange outputs.
